src/app/modules/item/components/chapter-detail/chapter-detail.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component } from '@angular/core';
+import { EMPTY } from 'rxjs';
+import { switchMap } from 'rxjs/operators';
+import { GetItemChildrenService, ItemChild } from '../../http-services/get-item-children.service';
+import { ItemDataSource } from '../../services/item-datasource.service';
+
+@Component({
+  selector: 'alg-chapter-detail',
+  templateUrl: './chapter-detail.component.html',
+  styleUrls: ['./chapter-detail.component.scss']
+})
+export class ChapterDetailComponent {
+
+  item$ = this.itemDataSource.item$;
```

Comment:
The item data source is actually dedicated to communication between the item-by-id page and its "route children" (in router-outlet) as it is the only way to pass information between these. For classical components, it is preferable to use the classical Angular communication between a component and its subcomponents: inputs.

Then, there are actually two possible way to pass the input:
- either you pass the observable (of state or item): it means the subcomponent needs to know what to do when its parent is in loading/error to hide itself (here in practice, nothing as its parents do not display it when in error/loading, but the child does not know that) and what to do with this stream of items where some items are not suitable for this component.
- either you pass the item itself in case it has to fetch its children.

I think the second option is cleaner (in terms of software design) for subcomponents of item-content. In practice, it means that at init and onchange, the previous service subscription should be be cancelled if any, the state should be to set to loading and then call the service. (with a true reloadData probably)